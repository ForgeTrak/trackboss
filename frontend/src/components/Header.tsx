import React, { useContext } from 'react';
import {
    Heading,
    Flex,
    Spacer,
    Box,
    Button,
    Text,
} from '@chakra-ui/react';

import { useNavigate } from 'react-router-dom';
import HamburgerMenu from './HamburgerMenu';
import { UserContext } from '../contexts/UserContext';

interface pageProps {
    title: string;
    activeButtonId: number;
}

export default function Header(props:pageProps) {
    const navigate = useNavigate();
    const navigateToMembers = () => {
        const path = '/members';
        navigate(path);
    };

    const { state, update } = useContext(UserContext);

    return (
        <div>
            <Flex bg="white" boxShadow="lg" padding="6">
                <HamburgerMenu activeButtonId={props.activeButtonId} />
                <Spacer />
                <Box>
                    <Heading pr={90} size="3xl">{props.title}</Heading>
                </Box>
                <Spacer />
            </Flex>
            {
                state.storedUser !== undefined && (
                    <Flex height={50} backgroundColor="red">
                        <Text
                            mt={2}
                            ml={10}
                            fontStyle="bold"
                            color="white"
                            fontSize="xl"
                        >
                            Currently acting as :
                            {' '}
                            { `${state.user?.firstName} ${state.user?.lastName}`}
                        </Text>
                        <Spacer />
                        <Button
                            mt={1.5}
                            ml={15}
                            mr={10}
                            size="md"
                            variant="outline"
                            color="white"
                            onClick={
                                () => {
                                    navigateToMembers();
                                    const originalUser = state.storedUser;
                                    // Return to the original user, and clear the storedUser
                                    update({
                                        loggedIn: true,
                                        token: state.token,
                                        user: originalUser,
                                        storedUser: undefined,
                                    });
                                }
                            }
                        >
                            RETURN TO YOUR PROFILE
                        </Button>
                    </Flex>
                )
            }
        </div>
    );
}
