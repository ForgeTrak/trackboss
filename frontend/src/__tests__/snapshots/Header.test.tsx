import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import Header from '../../components/Header';

it("renders correctly for Dashboard", () => {
    const header = renderer.create(
        <BrowserRouter>
            <Header title='Dashboard' activeButtonId={1}/>
        </BrowserRouter>
    ).toJSON();
    
    expect(header).toMatchSnapshot();
});

it("renders correctly for Calendar", () => {
    const header = renderer.create(
        <BrowserRouter>
            <Header title='Calendar' activeButtonId={2}/>
        </BrowserRouter>
    ).toJSON();
    
    expect(header).toMatchSnapshot();
});

it("renders correctly for Member List", () => {
    const header = renderer.create(
        <BrowserRouter>
            <Header title='Members' activeButtonId={3}/>
        </BrowserRouter>
    ).toJSON();
    
    expect(header).toMatchSnapshot();
});

it("renders correctly for Settings", () => {
    const header = renderer.create(
        <BrowserRouter>
            <Header title='General Info' activeButtonId={4}/>
        </BrowserRouter>
    ).toJSON();
    
    expect(header).toMatchSnapshot();
});
