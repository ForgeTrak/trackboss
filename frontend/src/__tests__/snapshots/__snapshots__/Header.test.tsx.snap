// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly for Calendar 1`] = `
<div>
  <div
    className="css-108gr1"
  >
    <div
      className="chakra-stack css-84zodg"
    >
      <button
        aria-label="Back"
        className="chakra-button css-1izmcir"
        onClick={[Function]}
        type="button"
      >
        <svg
          aria-hidden={true}
          fill="currentColor"
          focusable={false}
          height="lg"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 512 512"
          width="lg"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M217.9 256L345 129c9.4-9.4 9.4-24.6 0-33.9-9.4-9.4-24.6-9.3-34 0L167 239c-9.1 9.1-9.3 23.7-.7 33.1L310.9 417c4.7 4.7 10.9 7 17 7s12.3-2.3 17-7c9.4-9.4 9.4-24.6 0-33.9L217.9 256z"
          />
        </svg>
      </button>
      <div>
        <button
          aria-label="Menu"
          className="chakra-button css-4exw4h"
          onClick={[Function]}
          type="button"
        >
          <svg
            aria-hidden={true}
            fill="currentColor"
            focusable={false}
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 1024 1024"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M904 160H120c-4.4 0-8 3.6-8 8v64c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-64c0-4.4-3.6-8-8-8zm0 624H120c-4.4 0-8 3.6-8 8v64c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-64c0-4.4-3.6-8-8-8zm0-312H120c-4.4 0-8 3.6-8 8v64c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-64c0-4.4-3.6-8-8-8z"
            />
          </svg>
        </button>
      </div>
    </div>
    <div
      className="css-17xejub"
    />
    <div
      className="css-0"
    >
      <h2
        className="chakra-heading css-10o0cdp"
      >
        Calendar
      </h2>
    </div>
    <div
      className="css-17xejub"
    />
  </div>
</div>
`;

exports[`renders correctly for Dashboard 1`] = `
<div>
  <div
    className="css-108gr1"
  >
    <div
      className="chakra-stack css-84zodg"
    >
      <button
        aria-label="Back"
        className="chakra-button css-1izmcir"
        onClick={[Function]}
        type="button"
      >
        <svg
          aria-hidden={true}
          fill="currentColor"
          focusable={false}
          height="lg"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 512 512"
          width="lg"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M217.9 256L345 129c9.4-9.4 9.4-24.6 0-33.9-9.4-9.4-24.6-9.3-34 0L167 239c-9.1 9.1-9.3 23.7-.7 33.1L310.9 417c4.7 4.7 10.9 7 17 7s12.3-2.3 17-7c9.4-9.4 9.4-24.6 0-33.9L217.9 256z"
          />
        </svg>
      </button>
      <div>
        <button
          aria-label="Menu"
          className="chakra-button css-4exw4h"
          onClick={[Function]}
          type="button"
        >
          <svg
            aria-hidden={true}
            fill="currentColor"
            focusable={false}
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 1024 1024"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M904 160H120c-4.4 0-8 3.6-8 8v64c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-64c0-4.4-3.6-8-8-8zm0 624H120c-4.4 0-8 3.6-8 8v64c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-64c0-4.4-3.6-8-8-8zm0-312H120c-4.4 0-8 3.6-8 8v64c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-64c0-4.4-3.6-8-8-8z"
            />
          </svg>
        </button>
      </div>
    </div>
    <div
      className="css-17xejub"
    />
    <div
      className="css-0"
    >
      <h2
        className="chakra-heading css-10o0cdp"
      >
        Dashboard
      </h2>
    </div>
    <div
      className="css-17xejub"
    />
  </div>
</div>
`;

exports[`renders correctly for Member List 1`] = `
<div>
  <div
    className="css-108gr1"
  >
    <div
      className="chakra-stack css-84zodg"
    >
      <button
        aria-label="Back"
        className="chakra-button css-1izmcir"
        onClick={[Function]}
        type="button"
      >
        <svg
          aria-hidden={true}
          fill="currentColor"
          focusable={false}
          height="lg"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 512 512"
          width="lg"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M217.9 256L345 129c9.4-9.4 9.4-24.6 0-33.9-9.4-9.4-24.6-9.3-34 0L167 239c-9.1 9.1-9.3 23.7-.7 33.1L310.9 417c4.7 4.7 10.9 7 17 7s12.3-2.3 17-7c9.4-9.4 9.4-24.6 0-33.9L217.9 256z"
          />
        </svg>
      </button>
      <div>
        <button
          aria-label="Menu"
          className="chakra-button css-4exw4h"
          onClick={[Function]}
          type="button"
        >
          <svg
            aria-hidden={true}
            fill="currentColor"
            focusable={false}
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 1024 1024"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M904 160H120c-4.4 0-8 3.6-8 8v64c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-64c0-4.4-3.6-8-8-8zm0 624H120c-4.4 0-8 3.6-8 8v64c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-64c0-4.4-3.6-8-8-8zm0-312H120c-4.4 0-8 3.6-8 8v64c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-64c0-4.4-3.6-8-8-8z"
            />
          </svg>
        </button>
      </div>
    </div>
    <div
      className="css-17xejub"
    />
    <div
      className="css-0"
    >
      <h2
        className="chakra-heading css-10o0cdp"
      >
        Members
      </h2>
    </div>
    <div
      className="css-17xejub"
    />
  </div>
</div>
`;

exports[`renders correctly for Settings 1`] = `
<div>
  <div
    className="css-108gr1"
  >
    <div
      className="chakra-stack css-84zodg"
    >
      <button
        aria-label="Back"
        className="chakra-button css-1izmcir"
        onClick={[Function]}
        type="button"
      >
        <svg
          aria-hidden={true}
          fill="currentColor"
          focusable={false}
          height="lg"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 512 512"
          width="lg"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M217.9 256L345 129c9.4-9.4 9.4-24.6 0-33.9-9.4-9.4-24.6-9.3-34 0L167 239c-9.1 9.1-9.3 23.7-.7 33.1L310.9 417c4.7 4.7 10.9 7 17 7s12.3-2.3 17-7c9.4-9.4 9.4-24.6 0-33.9L217.9 256z"
          />
        </svg>
      </button>
      <div>
        <button
          aria-label="Menu"
          className="chakra-button css-4exw4h"
          onClick={[Function]}
          type="button"
        >
          <svg
            aria-hidden={true}
            fill="currentColor"
            focusable={false}
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 1024 1024"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M904 160H120c-4.4 0-8 3.6-8 8v64c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-64c0-4.4-3.6-8-8-8zm0 624H120c-4.4 0-8 3.6-8 8v64c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-64c0-4.4-3.6-8-8-8zm0-312H120c-4.4 0-8 3.6-8 8v64c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-64c0-4.4-3.6-8-8-8z"
            />
          </svg>
        </button>
      </div>
    </div>
    <div
      className="css-17xejub"
    />
    <div
      className="css-0"
    >
      <h2
        className="chakra-heading css-10o0cdp"
      >
        General Info
      </h2>
    </div>
    <div
      className="css-17xejub"
    />
  </div>
</div>
`;
